src/layers/ethernet/mod.rs

```diff
@@ -14,34 +14,55 @@ use crate::{Layer, Packet, ENCAP_TYPE_ETH};
 
 pub const ETH_HEADER_LENGTH: usize = 14_usize;
 
+/// A Map maintaining EtherType -> Creator fns for Layer Creators of L3 Layers.
+///
+/// The creator function simply creates a `default` L3 struct that implements the dissector
+/// for the Layer.
 pub fn get_ethertypes_map() -> &'static RwLock<HashMap<EtherType, LayerCreatorFn>> {
-    /// A Map maintaining EtherType -> Creator fns for Layer Creators of L3 Layers.
-    ///
-    /// The creator function simply creates a `default` L3 struct that implements the dissector
-    /// for the Layer.
-    pub(crate) static ETHERTYPES_MAP: OnceLock<RwLock<HashMap<EtherType, LayerCreatorFn>>> =
-        OnceLock::new();
+    static ETHERTYPES_MAP: OnceLock<RwLock<HashMap<EtherType, LayerCreatorFn>>> = OnceLock::new();
     ETHERTYPES_MAP.get_or_init(|| RwLock::new(HashMap::new()))
 }
 
+/// A Map maintaining String -> EtherType of L3 Layers.
+pub fn get_inv_ethertypes_map() -> &'static RwLock<HashMap<String, EtherType>> {
```

Comment:
May be this function is gated by `cfg(packet_sculpting)` ? Because this map and API is not used outside sculpting.